### PR TITLE
Merge test-project into playground

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

Merged test-project with playground.

- Moved all test-project tests into the `<playground>/Assets/Tests/` directory.
- No longer check-in generated code.
- Moved exhaustive schema out of test-project package and into playground schema.
- Removed redundant dependency test packages.
- Removed redundant modular codegen unit tests.

Includes drive-by disabling the flaky locator tests on mac.

#### Tests

- All tests pass through test runner UI.
- CI passes.

#### Documentation

n/a

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Assets/Tests/EditmodeTests/Utility/EntityTemplateTests.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Tests/EditmodeTests/Utility/EntityTemplateTests.cs:1]: This is an odd change. BOM?

File: workers/unity/Assets/Resources.meta [hunk 2]
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
-guid: 82256c4f21075e640a64c98a803fe7a2
+guid: 5bb31678f3981fd43bd427dbe649978a
```
Comment (zeroZshadow) [inline workers/unity/Assets/Resources.meta:2]: What happened here?

File: workers/unity/Assets/Tests/EditmodeTests/Utility/EntityTemplateTests.cs (hunk 1 above)
Comment (paulbalaji) [reply workers/unity/Assets/Tests/EditmodeTests/Utility/EntityTemplateTests.cs:1]: Probably - dotnet-format did it

File: workers/unity/Assets/Resources.meta (hunk 2 above)
Comment (paulbalaji) [reply workers/unity/Assets/Resources.meta]: I'm guessing Unity decided to give it a different guid now that it's in a different project? 

Good callout though, as this should be with the playmode tests

File: workers/unity/Assets/Tests/EditmodeTests/Subscriptions/SubscriptionAggregateCouplingTests.cs [hunk 3]
```diff
@@ -80,8 +80,10 @@ private static class TestMonoBehaviours
         {
             public abstract class Base<T> : MonoBehaviour
             {
+#pragma warning disable 649
```
Comment (austinkelmore) [inline workers/unity/Assets/Tests/EditmodeTests/Subscriptions/SubscriptionAggregateCouplingTests.cs:83]: What's the text for this warning? Maybe put it in a comment so it's a bit more clear what this is or why we're disabling it?

File: workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Connection/LocatorFlowTests.cs [hunk 4]
```diff
@@ -54,6 +54,7 @@ public void LocatorFlow_dev_auth_fails_if_no_deployment_running()
             Assert.IsInstanceOf<AuthenticationFailedException>(aggregateException.InnerExceptions[0]);
         }
 
+#if !UNITY_EDITOR_OSX
```
Comment (austinkelmore) [inline workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Connection/LocatorFlowTests.cs:57]: What's the reason we're only disabling this on OSX and not other platforms?

Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Connection/LocatorFlowTests.cs:57]: These tests flake a fair bit on Mac. It's an extension of this PR: https://github.com/spatialos/gdk-for-unity/pull/1361

Comment (austinkelmore) [reply workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Connection/LocatorFlowTests.cs:57]: Mind putting a comment there so it's easy to tell? Doesn't have to link to the PR, but just knowing why would be nice.

File: workers/unity/Assets/Tests/EditmodeTests/Subscriptions/SubscriptionAggregateCouplingTests.cs (hunk 3 above)
Comment (paulbalaji) [reply workers/unity/Assets/Tests/EditmodeTests/Subscriptions/SubscriptionAggregateCouplingTests.cs:83]: It'll say something like
```
Assets\Tests\EditmodeTests\Subscriptions\SubscriptionAggregateCouplingTests.cs(87,49): warning CS0649: Field 'SubscriptionAggregateCouplingTests.TestMonoBehaviours.Base<T>.Reader' is never assigned to, and will always have its default value null
```

Not sure we need a comment as we disable these warnings in several other places already

File: workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Connection/LocatorFlowTests.cs (hunk 4 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Connection/LocatorFlowTests.cs:57]: Sure, will add

**Pull request merged**